security/pkg/k8s/secret/secret_util.go

```diff
@@ -30,18 +32,10 @@ const (
 	privateKeyID = "key.pem"
 	// rootCertID is the ID/name for the CA root certificate file.
 	rootCertID = "root-cert.pem"
-	// serviceAccountNameAnnotationKey is the key to specify corresponding service account in the annotation of K8s secrets.
-	serviceAccountNameAnnotationKey = "istio.io/service-account.name"
 )
 
 // BuildSecret returns a secret struct, contents of which are filled with parameters passed in.
-func BuildSecret(saName, scrtName, namespace string, certChain, privateKey, rootCert, caCert, caPrivateKey []byte, secretType v1.SecretType) *v1.Secret {
-	var ServiceAccountNameAnnotation map[string]string
-	if saName == "" {
-		ServiceAccountNameAnnotation = nil
-	} else {
-		ServiceAccountNameAnnotation = map[string]string{serviceAccountNameAnnotationKey: saName}
```

Comment:
I think someone had concerns with this. IMO its fine, this is dead code since saName is always `""`